src/hooks/useMultiSelect.ts

```diff
@@ -0,0 +1,87 @@
+import { useState } from "react";
+import {
+  MultiSelectItem,
+  SelectedItems,
+} from "../components/MultiSelect/MultiSelectTypes";
+
+export default function useMultiSelect(multiSelectId, items) {
+  const [selectedItems, setSelectedItems] = useState<SelectedItems | {}>({});
+
+  function handleChange(itemId: string) {
+    let itemIds;
+    // Check if the id already exists in the state.
+    if (selectedItems[multiSelectId] !== undefined) {
+      let itemIdExists =
+        selectedItems[multiSelectId].items.indexOf(itemId) > -1;
+      // Make a copy of the existing items array.
+      itemIds = selectedItems[multiSelectId].items.slice();
+      // If id exists, remove it from the array.
+      if (itemIdExists) {
+        itemIds = itemIds.filter((id) => id !== itemId);
+      } else {
+        // Add it to the array, but modify the copy, not the original.
+        itemIds.push(itemId);
+      }
+    } else {
+      itemIds = [];
+      itemIds.push(itemId);
+    }
+
+    setSelectedItems({
+      ...selectedItems,
+      [multiSelectId]: {
+        items: itemIds,
+      },
+    });
+  }
+
+  function handleMixedStateChange(parentId: string) {
+    // Build an array of child items.
+    let childItems = [];
+    items.map((item) => {
```

Comment:
For the `.map` function, the callback function expects a value to be returned and it also returns a new array and it's not being assigned to any variable. It'd be better to use `.forEach` in this case and on line 43 as well.